src/main/java/org/craftercms/deployer/impl/processors/aws/S3SyncProcessor.java

```diff
@@ -117,24 +119,24 @@ protected ChangeSet doExecute(final Deployment deployment, final ProcessorExecut
     /**
      * Performs the upload of the given files.
      * @param client AWS S3 client
-     * @param files list of files to upload
+     * @param paths list of files to upload
      * @throws DeployerException if there is any error reading or uploading the files
      */
-    protected void uploadFiles(AmazonS3 client, List<String> files) throws DeployerException {
-        logger.info("Uploading {} files", files.size());
-        for(String file : files) {
-            logger.debug("Uploading file: {}", file);
-            try {
-                File localFile = new File(localRepoUrl, file);
-                PutObjectRequest request = new PutObjectRequest(s3Url.getBucket(), getS3Key(file), localFile);
-                client.putObject(request);
-            } catch (Exception e) {
-                logger.error("Error uploading file " + file, e);
-                throw new DeployerException("Error uploading file " + file, e);
-            }
-            logger.debug("Uploaded file: {}", file);
+    protected void uploadFiles(AmazonS3 client, List<String> paths) throws DeployerException {
+        logger.info("Uploading {} files", paths.size());
+        TransferManager transferManager = TransferManagerBuilder.standard().withS3Client(client).build();
```

Comment:
Do we have to build the TransferManager every time? Does the AmazonS3 client change on every call?

Reply:
yes, we are building a new client for each execution